src/search/pruning/null_pruning_method.h

```diff
@@ -9,7 +9,6 @@ class NullPruningMethod : public PruningMethod {
     virtual void initialize(const std::shared_ptr<AbstractTask> &) override;
     virtual void prune_operators(const State &,
                                  std::vector<OperatorID> &) override {}
-    virtual void print_statistics() const override {}
```

Comment:
You are right. What do you think is the better way of addressing this: making `print_statistics` `virtual` so that the null pruning method can override it or not printing any pruning statistics if there was no pruning? From an experiments perspective, I would prefer to always have the same output, also if there was no pruning with a non-null pruning method. But overriding to suppress doesn't seem that great either.

Reply:
Do we need the `NullPruningMethod` at all? It has been a long time since we implemented this and I don't quite remember why we added it. But working on the option parser now, I saw there is an option `OptionParser::NONE` that can be used as a default value for optional arguments without a default value. It could be that we just added `NullPruningMethod` because we were not aware of this option and did not know how to specify a default value for the option. The downside of using `NONE` is that within the search, the pruning method would then potentially `nullptr` and you'd have to add checks for that. Silvan, I saw this was used for label reduction in M&S, so you probably have experience with the trade-off.

Reply:
Good point. I think that the search calls the pruning method exactly once, so this would require a single additional if-clause. Definitely good trade-off.